libsycl/include/sycl/detail/config.hpp

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
```

Comment:
Is it expected that the files in `detail`, and in particular, `detail/config.hpp`, will *always* be included via a path that includes `sycl` such as `#include <sycl/detail/config.hpp>`? If so, the file naming here seems fine to me. Otherwise, we might want to do something like what libcxx, libcxxabi, and libunwind do and name these files with a project specific prefix. For example, libcxxabi has a `__cxxabi_config.h` file and libunwind has a `__libunwind_config.h` one (libcxx has a `__config` file under a `c++/v1` path). The concern is that `#include <detail/config.hpp>` could potentially conflict with header files provided by some other project used in an application.

Reply:
we always use "sycl/*" for either detail or API headers.
Example is https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/accessor.hpp#L14

Reply:
Great, thank you. As long as we always do that (and require users to do so as well), then I have no concerns about these names.